pkg/controller/master-controller-manager/rbac/mapper_test.go

```diff
@@ -210,3 +215,181 @@ func TestGenerateVerbsForResources(t *testing.T) {
+		if !reflect.DeepEqual(actualVerbs, tc.expectedVerbs) || (tc.wantErr && actualError == nil) || (!tc.wantErr && actualError != nil) {
+			t.Errorf("Test: %s - generateVerbsForNamespacedResource(%q, %q, %q) = %v, %v; expected %v, wantErr: %v", tc.name, tc.groupName, tc.resourceKind, tc.namespace, actualVerbs, actualError, tc.expectedVerbs, tc.wantErr)
+		}
+	}
+}
+
+func TestGenerateVerbsForNamedResourceInNamespace(t *testing.T) {
+	tests := []struct {
+		name          string
+		groupName     string
+		resourceKind  string
+		namespace     string
+		expectedVerbs []string
+		wantErr       bool
+	}{
+		// Namespace and kind acceptance cases
+		{
+			name:          "Owner can get, update, and delete secrets in saSecretsNamespaceName",
+			groupName:     OwnerGroupNamePrefix + "-group",
+			resourceKind:  secretV1Kind,
+			namespace:     saSecretsNamespaceName,
+			expectedVerbs: []string{"get", "update", "delete"},
+			wantErr:       false,
+		},
+		{
+			name:          "Project Manager can get, update, and delete secrets in KubeOneNamespacePrefix namespace",
+			groupName:     ProjectManagerGroupNamePrefix + "-group",
+			resourceKind:  secretV1Kind,
+			namespace:     resources.KubeOneNamespacePrefix + "some-namespace",
+			expectedVerbs: []string{"get", "update", "delete"},
+			wantErr:       false,
+		},
+		{
+			name:          "Unsupported Namespace for secrets",
+			groupName:     OwnerGroupNamePrefix + "-group",
+			resourceKind:  secretV1Kind,
+			namespace:     "unsupported-namespace",
+			expectedVerbs: nil,
+			wantErr:       true,
+		},
+		{
+			name:          "Unsupported Kind for secrets",
+			groupName:     OwnerGroupNamePrefix + "-group",
+			resourceKind:  "unsupported-kind",
+			namespace:     saSecretsNamespaceName,
+			expectedVerbs: nil,
+			wantErr:       true,
+		},
+
+		// CBSL in KubermaticNamespace
+		{
+			name:          "Owner can fully manage CBSL in KubermaticNamespace",
+			groupName:     OwnerGroupNamePrefix + "-group",
+			resourceKind:  kubermaticv1.ClusterBackupStorageLocationKind,
+			namespace:     resources.KubermaticNamespace,
+			expectedVerbs: []string{"get", "list", "create", "patch", "update", "delete"},
+			wantErr:       false,
+		},
+		// Unknown Group
+		{
+			name:          "Unknown Group",
+			groupName:     "unknown-group",
+			resourceKind:  secretV1Kind,
+			namespace:     resources.KubermaticNamespace,
+			expectedVerbs: nil,
+			wantErr:       false,
+		},
+	}
+
+	for _, tc := range tests {
+		actualVerbs, actualError := generateVerbsForNamedResourceInNamespace(tc.groupName, tc.resourceKind, tc.namespace)
+		if !reflect.DeepEqual(actualVerbs, tc.expectedVerbs) || (tc.wantErr && actualError == nil) || (!tc.wantErr && actualError != nil) {
+			t.Errorf("Test: %s - generateVerbsForNamedResourceInNamespace(%q, %q, %q) = %v, %v; expected %v, wantErr: %v", tc.name, tc.groupName, tc.resourceKind, tc.namespace, actualVerbs, actualError, tc.expectedVerbs, tc.wantErr)
+		}
```

Comment:
This looks like it's re-implementing `t.Run(name, func(t *testing.T) { .... })`, especially looking at the beginning of the Errorf message. :)

Reply:
I'm ready to take my chances here and go with this. @moelsayed maybe you can take this as a minor todo for post-release cleanup.